angr/exploration_techniques/loop_seer.py

```diff
@@ -143,13 +120,30 @@ def step(self, simgr, stash='active', **kwargs):
                 header = loop.entry.addr
                 exits = [e[1].addr for e in loop.break_edges]
 
-                state.loop_data.trip_counts[header].append(0)
+                state.loop_data.back_edge_trip_counts[header].append(0)
+                state.loop_data.header_trip_counts[header].append(0)
                 state.loop_data.current_loop.append((loop, exits))
 
         simgr.step(stash=stash, **kwargs)
 
         return simgr
 
-    def normalized_step(self, state):
+    def _normalized_step(self, state):
         node = self.cfg.get_any_node(state.addr)
         return state.step(num_inst=len(node.instruction_addrs) if node is not None else None)
+
+    def _get_function(self, func):
+        if type(func) is str:
+            f = self.cfg.kb.functions.function(name=func)
+            if f is None:
+                l.warning("Function '%s' doesn't exist in the CFG. Skipping...", func)
+
+        elif type(func) is int:
+            f = self.cfg.kb.functions.function(addr=func)
+            if f is None:
+                l.warning("Function at 0x%x doesn't exist in the CFG. Skipping...", func)
+
+        elif type(func) is Function:
+            f = func
+
+        return [] if f is None else [f]
```

Comment:
Why returning either an empty list or a list of only one element? How about returning `None` or the element itself?